### PR TITLE
Update configuration documentation

### PR DESCRIPTION
**Pull request by jameshfisher**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,42 +146,31 @@ If it's not `healthy` you should test your logstash configuration.
...
+The Logstash framework is configured at the time that the scheduler is started. Each configuration option can be passed in a large number of ways:
+
+1. Command-line arguments, e.g. `java -jar logstash-mesos-scheduler.jar --logstash.heapSize=64`
+2. Environment variables, e.g. `LOGSTASH_HEAP_SIZE=64 java -jar logstash-mesos-scheduler.jar`
+3. A properties file, e.g. `echo 'logstash.heap-size=64' > ./application.properties && java -jar logstash-mesos-scheduler.jar` 
+4. ...
+
+| Command-line argument            | Environment variable           | What it does                                                                                                               |
+| -------------------------------- | ------------------------------ | -------------------------------------------------------------------------------------------------------------------------- |
+| `--zk-url U`                     | `ZK_URL=U`                     | The Logstash framework will find Mesos using ZooKeeper at URL `U`, which must be in the format `zk://host:port/zkNode,...` |
```
Comment (mwl) [inline README.md]: For consistency I think we should equals-notation, `--key=val`. Has many upsides, like making order irrelevant and supports Docker's array `CMD`

Comment (jhftrifork) [inline README.md]: It depends on how this `@ConfigurationProperties` thing works. I couldn't find anything in the documentation which said how it parses command line arguments, so I guessed

File: README.md [hunk 2]
```diff
@@ -146,42 +146,31 @@ If it's not `healthy` you should test your logstash configuration.
...
+| `--zk-timeout T`                 | `ZK_TIMEOUT=T`                 | The Logstash framework will wait `T` milliseconds for ZooKeeper to respond before assuming that the session has timed out  |
+| `--framework-name N`             | `FRAMEWORK_NAME=N`             | The Logstash framework will show up in the Mesos Web UI with name `N`, and the ZK state will be rooted at znode `N`        |
+| `--webserver-port P`             | `WEBSERVER_PORT=P`             | The scheduler will listen on TCP port `P` FIXME and host what on it? Not clear from code                                   |
```
Comment (jhftrifork) [inline README.md]: @mwl Do you know what this option actually does? Afaics it just does `new JettyEmbeddedServletContainerFactory(P)` and then discards the factory

File: README.md [hunk 3]
```diff
@@ -146,42 +146,31 @@ If it's not `healthy` you should test your logstash configuration.
...
+| `--user U`                       | `USER=U`                       | Logstash tasks will be launched with Unix user `U`                                                                         |
+| `--logstash.heap-size N`         | `LOGSTASH_HEAP_SIZE=N`         | The Logstash program will be started with `LS_HEAP_SIZE=N` FIXME what does this actually do                                | 
+| `--logstash.elasticsearch-url U` | `LOGSTASH_ELASTICSEARCH_URL=U` | If present, Logstash will forward its logs to an Elasticsearch instance at domain and port `U`                             |
+| `--executor.cpus C`              | `EXECUTOR_CPUS=C`              | The Logstash framework will only accept resource offers with at least `C` CPUs. `C` must be a decimal greater than 0       | 
+| `--executor.heap-size H`         | `EXECUTOR_HEAP_SIZE=H`         | The memory allocation pool for the Logstash executor will be limited to `H` megabytes                                      |
+| `--enable.failover F`            | `ENABLE_FAILOVER=F`            | If `F` is `"true"`, all executors and tasks will remain running after this scheduler exits FIXME what's the format for `F` |
```
Comment (jhftrifork) [inline README.md]: @mwl do you know what format `@ConfigurationProperties` requires for boolean arguments?

File: README.md (hunk 1 above)
Comment (mwl) [inline README.md]: https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html
> When running in a new environment, an application.properties can be provided outside of your jar that overrides the `name`; and for one-off testing, you can launch with a specific command line switch (e.g. `java -jar app.jar --name="Spring"`).

Seems pretty clear to me

File: README.md (hunk 2 above)
Comment (mwl) [inline README.md]: I've removed it…

File: README.md (hunk 3 above)
Comment (mwl) [inline README.md]: Good question. Hopefully `--enable.failover` is enough? I'll check when I have a chance.

File: README.md [hunk 4]
```diff
@@ -146,42 +146,31 @@ If it's not `healthy` you should test your logstash configuration.
...
+| `--user=U`                       | `USER=U`                       | Logstash tasks will be launched with Unix user `U`                                                                         |
+| `--logstash.heap-size=N`         | `LOGSTASH_HEAP_SIZE=N`         | The Logstash program will be started with `LS_HEAP_SIZE=N` FIXME what does this actually do                                |
+| `--logstash.elasticsearch-url=U` | `LOGSTASH_ELASTICSEARCH_URL=U` | If present, Logstash will forward its logs to an Elasticsearch instance at domain and port `U`                             |
+| `--executor.cpus=C`              | `EXECUTOR_CPUS=C`              | The Logstash framework will only accept resource offers with at least `C` CPUs. `C` must be a decimal greater than 0       |
+| `--executor.heap-size=H`         | `EXECUTOR_HEAP_SIZE=H`         | The memory allocation pool for the Logstash executor will be limited to `H` megabytes                                      |
+| `--enable.failover=F`            | `ENABLE_FAILOVER=F`            | If `F` is `"true"`, all executors and tasks will remain running after this scheduler exits FIXME what's the format for `F` |
```
Comment (jhftrifork) [inline README.md]: @mwl a problem with the `--enable.failover` syntax would be that you can't override, say, application.properties to disable failover

Comment (jhftrifork) [inline README.md]: I'll look into what the syntax actually is.

File: README.md [hunk 5]
```diff
@@ -1,51 +1,109 @@
...
-- ☐ Support for streaming output from `docker log`
-- ☐ Support for arbitrary Logstash Plugins
-- ☐ Service Discovery (allow frameworks to discover the log service automatically, and configure themselves)
+* A Mesos cluster at version 0.22.1 or above.
+  Our scheduler and executors use version 0.25.0 of the Mesos API.
+
+* That Mesos cluster must have the `docker` containerizer enabled.
+
+* A Docker server must be running on every Mesos slave on port 2376,
+  to be used by the `docker` containerizer.
```
Comment (mwl) [inline README.md]: Mesos uses the docker socket. So not a requirement.

File: README.md [hunk 6]
```diff
@@ -1,51 +1,109 @@
...
+After satisfying the above requirements,
+start the Logstash framework by starting a Docker container from your `mesos/logstash-scheduler` image,
+like this:
+
+```bash
+> docker run mesos/logstash-scheduler \
+    --zk-url=zk://123.0.0.12:5181/logstash \
+    --zk-timeout=20000 \
+    --framework-name=logstash \
+    --webserver-port=10000 \
```
Comment (mwl) [inline README.md]: Deleted

File: README.md [hunk 7]
```diff
@@ -1,51 +1,109 @@
...
+    --user=root \
+    --logstash.heap-size=64 \
+    --logstash.elasticsearch-url=elasticsearch.service.consul:1234 \
+    --executor.cpus=0.5 \
+    --executor.heap-size=128 \
+    --enable.failover=false \
+    --enable.collectd=true \
+    --enable.syslog=true \
+    --enable.file=true \
+    --file.path='/var/log/*,/home/jhf/example.log'
```
Comment (mwl) [inline README.md]: It's `--executor.file-path=…`

**Comment by jhftrifork** [comment]
See #79

**Pull request closed without merging**